### ISSUE TITLE
[BUG] `deleteSts` and `updatePVC` get invoked unexpectedly when reading stale information from apiserver

### ISSUE THREAD

**Issue by srteam2020** [rabbitmq/cluster-operator#653]
## Describe the bug
We find that after a restart the controller could accidentally invoke `deleteSts` and `updatePVC`, which leads to unexpected statefulset deletion and a try to resize PVC. We find that the root cause is the staleness populated from the apiserver that makes the controller think the "desired" capacity (from the stale RabbitmqCluster spec) does not equal the current capacity in `needsPVCResize` and invokes `deleteSts` (in `expandPVC`) to delete the existing statefulset. This unexpected statefulset deletion can be fully avoided by checking whether the desired capacity is larger than the current capacity in `needsPVCResize`, given the size of PVC can only grow in one direction. If not, `expandPVC` should not be invoked because it indicates an invalid `desired` capacity  potentially caused by staleness in apiserver.

## To Reproduce
1. Create a RabbitmqCluster `rabbitmq-cluster1` in a HA k8s cluster with PVC sized 10Gi. The controller is talking to apiserver1.
2. Resize the PVC of `rabbitmq-cluster1` to 15 Gi. The controller will find the desired capacity is larger than the current one so it will delete the sts, enlarge the PVC and recreate the sts. Meanwhile, apiserver2 gets stuck at the view before resizing `rabbitmq-cluster1`'s PVC to 15Gi due to some temporary network issues.
3. The controller restarts after a node failure and talks to the stale apiserver2. Reading the stale information about `rabbitmq-cluster1` from apiserver2, the controller thinks the "desired" capacity is still 10Gi that is different from the current capacity (15Gi). The `needsPVCResize` will return true and the controller will delete the statefulset by calling `deleteSts`.

**Expected behavior**
The controller should be able to tell that the "desired" capacity shown in the RabbitmqCluster spec is invalid (stale) by checking if it is larger than the current capacity. Since the PVC size can only be enlarged (rather than reduced), any "desired" capacity value smaller than the current one is invalid (potentially caused by staleness in apiserver), and the controller should directly return with the error indicating staleness, instead of calling `deleteSts` in such case.

## Version and environment information
- RabbitMQ: 3.8.12
- RabbitMQ Cluster Operator: 4f13b9a942ad34fece0171d2174aa0264b10e947 (main branch)
- Kubernetes: 1.18.9
 
## Additional context
We are willing to issue a PR to help fix this issue. We can change the check
```
	if currentCapacity.Cmp(desiredCapacity) != 0 {
		return true, nil
	}
```
to
```
	if currentCapacity.Cmp(desiredCapacity) == -1 {
		return true, nil
	}
```
in `needsPVCResize` (`-1` indicates `desiredCapacity` is larger than `currentCapacity`). So that stale "desired" capacity will not trigger statefulset deletion any more.

**Comment by ChunyiLyu** [comment]
@srteam2020 You are right that the statefulSet shouldn't be deleted if we know that the following PVC resize would fail anyways. I'm creating a PR to fix it. Thanks for catching it.
